# Quality of Life Dataset Analysis

### Abstract

### Dataset description

The dataset belongs to the socio-economic domain and contains quality of life indicators for different countries between 2015 and 2024.

It includes information related to purchasing power, safety, healthcare, cost of living, pollution, climate conditions, and overall quality of life.

In [101]:
import pandas as pd

In [102]:
data = pd.read_csv("../data_sets/quality_of_life_indices_by_country.csv")

In [103]:
data.head()

,Rank,Country,Quality of Life Index,Purchasing Power Index,Safety Index,Health Care Index,Cost of Living Index,Property Price to Income Ratio,Traffic Commute Time Index,Pollution Index,Climate Index,Year
0,1,Switzerland,222.9,146.5,73.2,66.3,126.0,7.3,25.6,24.1,-,2015
1,2,Germany,195.9,111.8,71.5,75.6,76.3,6.6,31.1,28.3,-,2015
2,3,Sweden,193.9,110.9,57.9,76.3,82.9,9.2,26.9,15.1,-,2015
3,4,United States,192.5,126.1,50.0,67.8,76.5,2.6,36.0,31.4,-,2015
4,5,Finland,190.2,101.2,70.5,69.4,89.7,7.0,33.2,14.9,-,2015


In [104]:
data.shape

(1495, 12)

The dataset contains 12 fields.

There is one categorical field (Country), one ranking field (Rank), and ten variables describing different aspects of quality of life.

These variables include the overall Quality of Life Index, Purchasing Power Index, Safety Index, Health Care Index, Cost of Living Index, Property Price to Income Ratio, Traffic Commute Time Index, Pollution Index, Climate Index, and Year.

In [105]:
data.info()

<class 'pandas.DataFrame'>
RangeIndex: 1495 entries, 0 to 1494
Data columns (total 12 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   Rank                            1495 non-null   int64  
 1   Country                         1495 non-null   str    
 2   Quality of Life Index           1495 non-null   float64
 3   Purchasing Power Index          1495 non-null   float64
 4   Safety Index                    1495 non-null   float64
 5   Health Care Index               1495 non-null   float64
 6   Cost of Living Index            1495 non-null   float64
 7   Property Price to Income Ratio  1495 non-null   float64
 8   Traffic Commute Time Index      1495 non-null   float64
 9   Pollution Index                 1495 non-null   float64
 10  Climate Index                   1495 non-null   str    
 11  Year                            1495 non-null   str    
dtypes: float64(8), int64(1), str(3)
memory usage:

In [106]:
data.isna().sum()

Rank                              0
Country                           0
Quality of Life Index             0
Purchasing Power Index            0
Safety Index                      0
Health Care Index                 0
Cost of Living Index              0
Property Price to Income Ratio    0
Traffic Commute Time Index        0
Pollution Index                   0
Climate Index                     0
Year                              0
dtype: int64

In [107]:
data.duplicated().sum()

np.int64(0)

In [108]:
print("Non-numeric Climate Index values:", (data["Climate Index"] == "-").sum())

Non-numeric Climate Index values: 143


In [109]:
data["Year"].unique()

<StringArray>
[  '2015', '2015/2',   '2016', '2016/2',   '2017', '2017/2',   '2018',
 '2018/2',   '2019', '2019/2',   '2020', '2020/2',   '2021', '2021/2',
   '2022', '2022/2',   '2023', '2023/2',   '2024', '2024/2']
Length: 20, dtype: str

The dataset is generally of good quality. No missing values or duplicate rows were detected.

However, some inconsistencies were found. The Year column contains values such as "2015/2", while the Climate Index column contains 143 non-numeric entries represented by "-". Because of these inconsistencies, both columns are stored as strings instead of numeric data types. These issues will be corrected during the data cleaning stage.

### Descriptive statistics

In [110]:
statistics = pd.DataFrame({
    "Mean":[data["Quality of Life Index"].mean(),
            data["Purchasing Power Index"].mean(),
            data["Safety Index"].mean(),
            data["Health Care Index"].mean(),
            data["Cost of Living Index"].mean()],

    "Median":[
        data["Quality of Life Index"].median(),
        data["Purchasing Power Index"].median(),
        data["Safety Index"].median(),
        data["Health Care Index"].median(),
        data["Cost of Living Index"].median()],

    "Standard Deviation":
        [
        data["Quality of Life Index"].std(),
        data["Purchasing Power Index"].std(),
        data["Safety Index"].std(),
        data["Health Care Index"].std(),
        data["Cost of Living Index"].std()
        ],
},
    index=["Quality of Life Index",
           "Purchasing Power Index",
           "Safety Index",
           "Health Care Index",
           "Cost of Living Index"]
)

statistics.round(2)

,Mean,Median,Standard Deviation
Quality of Life Index,134.16,136.2,40.83
Purchasing Power Index,66.80,60.7,33.28
Safety Index,60.10,59.9,13.78
Health Care Index,65.21,66.4,10.15
Cost of Living Index,53.44,49.2,20.57


Among the selected indicators, the Quality of Life Index has the highest variability, while the Health Care Index shows the lowest variability.

The mean and median values are relatively close for all indicators, suggesting that the data does not contain many extreme values.


### Data cleanup

In [111]:
clean_data = data.drop(columns = "Climate Index")
clean_data.head()

,Rank,Country,Quality of Life Index,Purchasing Power Index,Safety Index,Health Care Index,Cost of Living Index,Property Price to Income Ratio,Traffic Commute Time Index,Pollution Index,Year
0,1,Switzerland,222.9,146.5,73.2,66.3,126.0,7.3,25.6,24.1,2015
1,2,Germany,195.9,111.8,71.5,75.6,76.3,6.6,31.1,28.3,2015
2,3,Sweden,193.9,110.9,57.9,76.3,82.9,9.2,26.9,15.1,2015
3,4,United States,192.5,126.1,50.0,67.8,76.5,2.6,36.0,31.4,2015
4,5,Finland,190.2,101.2,70.5,69.4,89.7,7.0,33.2,14.9,2015


Since the Climate Index column contains “-” for all observations from 2015, removing these values would result in losing an entire year of data. Therefore, the Climate Index column was removed from the dataset. Additionally, this indicator was not intended to be used in the correlation analysis or visualizations performed later in the project.

In [112]:
clean_data["Year"] = clean_data["Year"].str.replace("/2", "")
clean_data["Year"] = clean_data["Year"].astype(int)

display(clean_data["Year"].unique())
display(clean_data.dtypes)

array([2015, 2016, 2017, 2018, 2019, 2020, 2021, 2022, 2023, 2024])

Rank                                int64
Country                               str
Quality of Life Index             float64
Purchasing Power Index            float64
Safety Index                      float64
Health Care Index                 float64
Cost of Living Index              float64
Property Price to Income Ratio    float64
Traffic Commute Time Index        float64
Pollution Index                   float64
Year                                int64
dtype: object

In [113]:
clean_data = clean_data.groupby(["Country","Year"], as_index=False).mean()
clean_data.head(15)

,Country,Year,Rank,Quality of Life Index,Purchasing Power Index,Safety Index,Health Care Index,Cost of Living Index,Property Price to Income Ratio,Traffic Commute Time Index,Pollution Index
0,Albania,2024,70.5,102.00,35.75,54.50,49.25,41.45,14.10,36.70,77.10
1,Argentina,2015,48.0,84.90,71.50,37.05,75.00,67.65,12.25,42.25,57.75
2,Argentina,2016,37.5,137.90,68.10,36.80,73.90,59.70,12.65,40.75,57.30
3,Argentina,2017,37.0,129.65,54.60,36.95,72.75,54.70,11.95,38.40,54.20
4,Argentina,2018,41.0,131.10,60.85,38.15,70.90,45.05,11.90,41.90,54.05
5,Argentina,2019,49.0,121.75,53.50,37.20,69.60,35.90,19.25,42.55,52.10
6,Argentina,2020,55.0,114.75,45.30,37.95,69.25,32.10,24.30,43.05,50.65
7,Argentina,2021,59.5,107.40,31.15,36.45,68.80,36.60,24.55,42.90,50.85
8,Argentina,2022,63.5,107.75,32.15,35.90,69.15,33.40,24.40,43.35,51.05
9,Argentina,2023,59.5,110.60,35.35,35.85,68.60,30.75,22.50,44.00,51.15


The dataset contains semi-annual observations marked with the “/2” suffix. Since this project focuses on yearly analysis, the suffix was removed from the Year column and the values were converted to integers. Then, observations for the same country and year were combined by calculating the mean of the numerical indicators.

In [114]:
years_countries = clean_data.groupby("Country")["Year"].nunique()
years_countries.value_counts().sort_index()

Year
1     11
2      2
3      1
4      2
5      3
6      4
7      9
8      2
9      3
10    61
Name: count, dtype: int64

In [115]:
good_countries = years_countries[years_countries >= 7].index
clean_data = clean_data[clean_data["Country"].isin(good_countries)]
clean_data

,Country,Year,Rank,Quality of Life Index,Purchasing Power Index,Safety Index,Health Care Index,Cost of Living Index,Property Price to Income Ratio,Traffic Commute Time Index,Pollution Index
1,Argentina,2015,48.0,84.90,71.50,37.05,75.00,67.65,12.25,42.25,57.75
2,Argentina,2016,37.5,137.90,68.10,36.80,73.90,59.70,12.65,40.75,57.30
3,Argentina,2017,37.0,129.65,54.60,36.95,72.75,54.70,11.95,38.40,54.20
4,Argentina,2018,41.0,131.10,60.85,38.15,70.90,45.05,11.90,41.90,54.05
5,Argentina,2019,49.0,121.75,53.50,37.20,69.60,35.90,19.25,42.55,52.10
...,...,...,...,...,...,...,...,...,...,...,...
776,Vietnam,2020,73.5,87.00,27.55,54.10,57.85,38.20,21.65,30.00,86.25
777,Vietnam,2021,75.0,88.65,26.30,53.85,58.45,37.45,20.20,29.60,85.70
778,Vietnam,2022,79.5,88.15,28.80,54.20,59.00,36.55,22.30,29.75,85.55
779,Vietnam,2023,75.5,89.30,30.90,55.70,59.60,35.10,23.80,29.45,84.65


Some countries have data for only a few years. To ensure more reliable analysis, only countries with observations available for at least seven years were retained.

In [116]:
print(clean_data.shape)
display(clean_data.dtypes)

(716, 11)


Country                               str
Year                                int64
Rank                              float64
Quality of Life Index             float64
Purchasing Power Index            float64
Safety Index                      float64
Health Care Index                 float64
Cost of Living Index              float64
Property Price to Income Ratio    float64
Traffic Commute Time Index        float64
Pollution Index                   float64
dtype: object

After cleaning and filtering, the dataset contains 716 rows and 11 columns. All remaining columns have appropriate data types for further analysis.